File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task169.java

import java.util.*;

public class Task169 {
    private static class TreeNode {
        int val;
        int count;
        int leftSize;
        TreeNode left;
        TreeNode right;
        
        TreeNode(int val) {
            this.val = val;
            this.count = 1;
            this.leftSize = 0;
        }
    }
    
    public static List<Integer> countSmaller(int[] nums) {
        if (nums == null || nums.length == 0) {
            return new ArrayList<>();
        }
        
        Integer[] result = new Integer[nums.length];
        TreeNode root = null;
        
        for (int i = nums.length - 1; i >= 0; i--) {
            root = insert(root, nums[i], result, i, 0);
        }
        
        return Arrays.asList(result);
    }
    
    private static TreeNode insert(TreeNode node, int val, Integer[] result, int index, int preSum) {
        if (node == null) {
            node = new TreeNode(val);
            result[index] = preSum;
        } else if (node.val == val) {
            node.count++;
            result[index] = preSum + node.leftSize;
        } else if (node.val > val) {
            node.leftSize++;
            node.left = insert(node.left, val, result, index, preSum);
        } else {
            node.right = insert(node.right, val, result, index, preSum + node.leftSize + node.count);
        }
        return node;
    }
    
    public static void main(String[] args) {
        // Test case 1
        int[] nums1 = {5, 2, 6, 1};
        System.out.println("Test 1: " + countSmaller(nums1));
        
        // Test case 2
        int[] nums2 = {-1};
        System.out.println("Test 2: " + countSmaller(nums2));
        
        // Test case 3
        int[] nums3 = {-1, -1};
        System.out.println("Test 3: " + countSmaller(nums3));
        
        // Test case 4
        int[] nums4 = {1, 2, 3, 4, 5};
        System.out.println("Test 4: " + countSmaller(nums4));
        
        // Test case 5
        int[] nums5 = {5, 4, 3, 2, 1};
        System.out.println("Test 5: " + countSmaller(nums5));
    }
}

File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task169.py

class TreeNode:
    def __init__(self, val):
        self.val = val
        self.count = 1
        self.left_size = 0
        self.left = None
        self.right = None

def countSmaller(nums):
    if not nums:
        return []
    
    result = [0] * len(nums)
    root = None
    
    for i in range(len(nums) - 1, -1, -1):
        root = insert(root, nums[i], result, i, 0)
    
    return result

def insert(node, val, result, index, pre_sum):
    if node is None:
        node = TreeNode(val)
        result[index] = pre_sum
    elif node.val == val:
        node.count += 1
        result[index] = pre_sum + node.left_size
    elif node.val > val:
        node.left_size += 1
        node.left = insert(node.left, val, result, index, pre_sum)
    else:
        node.right = insert(node.right, val, result, index, pre_sum + node.left_size + node.count)
    return node

if __name__ == "__main__":
    # Test case 1
    nums1 = [5, 2, 6, 1]
    print("Test 1:", countSmaller(nums1))
    
    # Test case 2
    nums2 = [-1]
    print("Test 2:", countSmaller(nums2))
    
    # Test case 3
    nums3 = [-1, -1]
    print("Test 3:", countSmaller(nums3))
    
    # Test case 4
    nums4 = [1, 2, 3, 4, 5]
    print("Test 4:", countSmaller(nums4))
    
    # Test case 5
    nums5 = [5, 4, 3, 2, 1]
    print("Test 5:", countSmaller(nums5))
